### ISSUE TITLE
Support the pure characteristics model

### ISSUE THREAD

**Issue by jeffgortmaker** [jeffgortmaker/pyblp#26]
From Berry and Pakes (2007). This wouldn't require many changes to the API, but it would require new methods for solving the contraction mapping. Analytic Jacobians will likely be problematic as well.

**Comment by fpinter** [comment]
The "mixture of vertical models" method of Berry and Pakes (2007) requires that no two products within a market have the same price. It looks like this is true for CPUs ([Song](http://www.jstor.org/stable/25046314), [Nosko](http://faculty.chicagobooth.edu/workshops/marketing/past/PDF/nosko_jmp.pdf)) but it might not be true in all empirical settings.

You might have some luck with [a proposed method by Lixiong Li](http://arxiv.org/abs/1802.04444), and another grad student has reported some success using the MCMC approach of [Chernozhukov and Hong (2003)](http://www.sciencedirect.com/science/article/pii/S0304407603001003).

If you're willing to go with the Berry and Pakes assumptions: another student and I tried using MPEC, and it did pretty well in simulations if the starting values were good enough. We did spend some time with the Jacobians if you're interested in discussing further.
